cmd/snap-confine/mount-support.c

```diff
@@ -606,33 +601,17 @@ void sc_populate_mount_ns(struct sc_apparmor *apparmor, int snap_update_ns_fd,
 		char rootfs_dir[PATH_MAX] = { 0 };
 		sc_must_snprintf(rootfs_dir, sizeof rootfs_dir,
 				 "%s/%s/current/", SNAP_MOUNT_DIR,
-				 base_snap_name);
+				 sc_effective_base_snap_name(inv));
 		if (access(rootfs_dir, F_OK) != 0) {
```

Comment:
not necessarily for now but I wonder if computing rootfs_dir  and checking for it could also be moved early and carry that in `sc_invocation` instead or as well

Reply:
Ah interesting.

Since we already carry information that was not provided to snap-confine externally inside the invocation object we can certainly simplify some of the code (not just this one here) that cares about the upcoming root filesystem path (terminology is somewhat confusing since the actual rootfs contains more items and is constructed elsewhere).

I'm happy to do iterative passes to see what we can simplify, especially as I also plan to remove some of the code down the line, that would move to snap-update-ns'es existing capabilities.

Reply:
yes, I noticed this kind of code is in another places, and also finding fallbacks needs to do it as well